contracts/swap/factory.go

```diff
@@ -0,0 +1,123 @@
+package swap
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	contractFactory "github.com/ethersphere/go-sw3/contracts-v0-1-1/simpleswapfactory"
```

Comment:
I think it would be better to call it `chequebookFactory`

Reply:
Changed. Although that's also not ideal. Now the package is `chequebookFactory` but the type is `SimpleSwapFactory`.